core/search/sqlite/sqlite.go

```diff
@@ -0,0 +1,139 @@
+package sqlite
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"gorm.io/gorm/logger"
+
+	"github.com/FleekHQ/space-daemon/core/search"
+
+	"github.com/pkg/errors"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+)
+
+const DbFileName = "filesIndex.db"
+
+type sqliteSearchOption struct {
+	dbPath   string
+	logLevel logger.LogLevel
+}
+
+type Option func(o *sqliteSearchOption)
+
+// sqliteFilesSearchEngine is a files search engine that is backed by sqlite
+type sqliteFilesSearchEngine struct {
+	db   *gorm.DB
+	opts sqliteSearchOption
+}
+
+// Creates a new SQLite backed search engine for files and folders
+func NewSearchEngine(opts ...Option) *sqliteFilesSearchEngine {
+	searchOptions := sqliteSearchOption{
+		dbPath: fmt.Sprintf("~%c.fleek-space", os.PathSeparator),
+	}
+
+	for _, opt := range opts {
+		opt(&searchOptions)
+	}
+
+	return &sqliteFilesSearchEngine{
+		db:   nil,
+		opts: searchOptions,
+	}
+}
+
+func (s *sqliteFilesSearchEngine) Start() error {
+	dsn := fmt.Sprintf("%s%c%s", s.opts.dbPath, os.PathSeparator, DbFileName)
+	if db, err := gorm.Open(sqlite.Open(dsn), &gorm.Config{
+		Logger: logger.Default.LogMode(s.opts.logLevel),
+	}); err != nil {
+		return errors.Wrap(err, "failed to open database")
+	} else {
+		s.db = db
+	}
+
+	return s.db.AutoMigrate(&SearchIndexRecord{})
+}
+
+func (s *sqliteFilesSearchEngine) InsertFileData(ctx context.Context, data *search.InsertIndexRecord) (*search.IndexRecord, error) {
+	record := SearchIndexRecord{
+		ItemName:      data.ItemName,
+		ItemExtension: data.ItemExtension,
+		ItemPath:      data.ItemPath,
+		ItemType:      data.ItemPath,
+		BucketSlug:    data.BucketSlug,
+		DbId:          data.DbId,
+	}
+	result := s.db.Create(&record)
+
+	if result.Error != nil {
+		if strings.Contains(result.Error.Error(), "UNIQUE constraint failed") {
+			return nil, errors.New("a similar file has already been inserted")
+		}
+		return nil, result.Error
+	}
+
+	return modelToIndexRecord(&record), nil
+}
+
+func (s *sqliteFilesSearchEngine) DeleteFileData(ctx context.Context, data *search.DeleteIndexRecord) error {
+	stmt := s.db.Where(
+		"item_name = ? AND item_path = ? AND bucket_slug = ?",
+		data.ItemName,
+		data.ItemPath,
+		data.BucketSlug,
+	)
+	if data.DbId != "" {
+		stmt = stmt.Where("dbId = ?", data.DbId)
+	}
+
+	result := stmt.Delete(&SearchIndexRecord{})
+
+	return result.Error
+}
+
+func (s *sqliteFilesSearchEngine) QueryFileData(ctx context.Context, query string, limit int) ([]*search.IndexRecord, error) {
+	var records []*SearchIndexRecord
+	result := s.db.Where(
+		"item_name LIKE ? OR item_extension = ?",
```

Comment:
Should the extension be a LIKE too?

Reply:
Hmm....not sure, honestly, I feel the extension check might be redundant, since the item name already contains the  extension